### PR TITLE
Make sure root surface transformations survive resetting the matrix directly in Flow.

### PR DESCRIPTION
**Pull request by chinmaygarde**
This used to only be handled correctly for non-root layer backing stores. This
was mostly a side effect of the fact that we used recording canvases instead of
rendering directly into the backing store. We now use recording canvases
consistently.

Fixes b/143464703

### CODE REVIEW

**Comment by chinmaygarde** [comment]
xref b/143464703

**Comment by chinmaygarde** [comment]
Fixes https://github.com/flutter/flutter/issues/43732

File: shell/platform/embedder/embedder_external_view_embedder.cc [hunk 1]
```diff
@@ -95,24 +95,14 @@ void EmbedderExternalViewEmbedder::BeginFrame(SkISize frame_size,
     }
   }
 
-  // If there is no root render target, create one now. This will be accessed by
-  // the rasterizer before the submit call layer to access the surface surface
-  // canvas.
+  // If there is no root render target, create one now.
+  // TODO(chinmaygarde): This can now be moved to be later in the submit call.
```
Comment (amirh) [inline shell/platform/embedder/embedder_external_view_embedder.cc]: nit: If it's something you think we should do let's have a tracking bug and change the language from "can" to "should"

File: shell/platform/embedder/embedder_external_view_embedder.cc [hunk 2]
```diff
@@ -237,8 +227,15 @@ bool EmbedderExternalViewEmbedder::SubmitFrame(GrContext* context) {
     return false;
   }
 
+  // Copy the contents of the root picture recorder onto the root surface
+  // while making sure to take into account any surface transformations.
   if (auto root_canvas = root_render_target_->GetRenderSurface()->getCanvas()) {
+    root_canvas->setMatrix(pending_surface_transformation_);
+    root_canvas->clear(SK_ColorTRANSPARENT);
```
Comment (amirh) [inline shell/platform/embedder/embedder_external_view_embedder.cc]: Isn't this redundant for the root canvas? do you know if there's a performance cost for doing it twice?

File: shell/platform/embedder/fixtures/main.dart [hunk 3]
```diff
@@ -453,3 +453,37 @@ void can_receive_locale_updates() {
   };
   signalNativeTest();
 }
+
+@pragma('vm:entry-point')
+void verify_b143464703() {
```
Comment (amirh) [inline shell/platform/embedder/fixtures/main.dart]: Maybe include a link to a GitHub issue if you're referring a bug, this number is pretty mysterious for non googlers

File: shell/platform/embedder/embedder_external_view_embedder.cc (hunk 1 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder_external_view_embedder.cc]: Done and filed https://github.com/flutter/flutter/issues/43778

File: shell/platform/embedder/embedder_external_view_embedder.cc (hunk 2 above)
Comment (chinmaygarde) [reply shell/platform/embedder/embedder_external_view_embedder.cc]: No. The embedder may have unnecessary stuff on the render target that we need to get rid of.

File: shell/platform/embedder/fixtures/main.dart (hunk 3 above)
Comment (chinmaygarde) [reply shell/platform/embedder/fixtures/main.dart]: Done. There was no GitHub issue when I began work on this.

**Pull request merged**